Functions_and_Scope_Lab.js

```diff
@@ -0,0 +1,370 @@
+//
+//
+// Example:
+//
+// Input: `first(3)`
+//
+// Output: `[1, 2, 3]`
+//
+//
+// ## Question 8
+console.log("_____#8_____")
+//
+let counter1 = 0
+const fizzBuzz = (num) => {
+  for (let i = 0; i < num + 1; i++) {
+    counter1++
+   if (counter1 % 3 === 0 && counter1 % 5 === 0) {
+    console.log("FizzBuzz")
+  } else if (counter1 % 5 === 0) {
+    console.log("Buzz")
+  } else if (counter1 % 3 === 0) {
+    console.log("Fizz")
+  } else {
+    console.log(counter1)
+  }
+ }
+}
+
+fizzBuzz(20)
+
+
+// Write a function that logs the numbers from 1 to x, except:
+//
+// If the number if a multiple of 3, log `"Fizz"` instead of the number
+// If the number is a multiple of 5, log `"Buzz"` instead of the number
+// If the number is a multiple of 3 AND 5, log `"FizzBuzz"` instead of the number
+// Your function should take in one parameter: x (the number to count up to)
+//
+//
+// ## Question 9
+console.log("_____#9_____")
+//
+let forwardArr = [1, 2, 3, 4]
+
+const reverse = (arr) => {
+  let reverseArr = []
+  for (let i = arr.length - 1; i >= 0; i--) {
+  reverseArr.push(arr[i])
+ }
+ return reverseArr
+}
+
+let return9 = reverse(forwardArr)
+console.log(return9)
+
+// Write a function named `reverse` that takes an array as a parameter. The function should return an array in reverse order.
+//
+//
+// Example:
+//
+// Input: `reverse([1, 2, 3])`
+//
+// Output: `[3, 2, 1]`
+//
+//
+// ## Question 10
+console.log("_____#10_____")
+//
+// Write a function that prints out the most frequently occurring element in an array.
+//
+
+
+// Example:
+//
+let input2 = [2,1,4,2,5,2,3,23,3,3,2,2,2,24,56,2,2,5,46,2]
+const freqElementArr = (arr) => {
+  let counter = {}
+  let max = arr[0]
+  let maxCount = 1
+  for (let i = 0; i < arr.length; i++) {
+    let value = arr[i]
+    if (counter[value] == null) {
+      counter[value] = 1
+    } else {
+      counter[value]++
+    }
+    if (counter[value] > maxCount){
+      max = value
+      maxCount = counter[value]
+    }
+  }
+  return max
+}
+
+let return10 = freqElementArr(input2)
+console.log(return10)
+
+// NOTE: I'm INCREDIBLY lost on this one. The code doesn't brake, but it doesn't
+// return anything either.
+
+//
+// Output: `2`
+//
+//
+//
+// ## Question 11
+console.log("_____#11_____")
+//
+// Write a function that sums all the even numbers in an array.
+//
+// Example:
+//
+let input1 = [1,2,5,64,23,22,91,3]
+
+const sumOfEven = (arr) => {
+  let sum2 = 0
+  for(let i = 0; i < arr.length; i++) {
+    if (arr[i] % 2 === 0) {
+    sum2 += arr[i]
+    console.log(sum2);
+    }
+  }
+  return sum2
+}
+let return11 = sumOfEven(input1)
+console.log(return3)
+
+
+//
+// Output: `88`
+//
+//
+// ## Question 12
+console.log("_____#12_____")
+//
+// Write a function that flips an object.  All of the keys are now values and all of the values are now keys.
+//
+
+let spanishCountingObj = {
+  1: "Uno",
+  5: "Cinco"
+}
+const objKeyValSwap = (obj) => {
```

Comment:
There can be much simpler way of doing this rather than using 3 loops and an array. How can we do that?